sdk/storage/storage-blob-changefeed/test/blobchangefeedclient.spec.ts

```diff
@@ -127,7 +127,7 @@ describe("BlobChangeFeedClient", async () => {
       await iter.next();
       assert.fail("Should have been aborted.");
     } catch (err) {
-      assert.equal(err.name, "AbortError");
+      assert.equal((err as any).name, "AbortError");
```

Comment:
I'm using Version 4.3.5. 
If I didn't specify a type for err it would complain about err being `unknown`.

Also, I tried to check if `err` was an instance of `RestErrror` by adding this condition before the asserts:
```
if (!(err instanceof RestError)) {
        throw new Error("Error is not recognized");
      }
```
However, some tests started to fail because their errors were not `RestError`, I could not figure out what error type they were, so I cast them as `any`.

Reply:
I believe the compiler starts reporting this error in 4.4. See https://github.com/azure/azure-sdk-for-js/issues/17776.

I wonder whether you only see this error in VS Code when recent version of TypeScript compiler is used to power the IDE. I wouldn't expect `rushx build` to report error for this currently

Reply:
I see, I do get the errors on VS Code, according to [this document](https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-newer-typescript-versions) it ships with the latest stable version (4.5.2 in my case)

What would be your suggestion on the unknown `err` vars? Should I discard the changes or leave them?

Reply:
You could assert them to a real type instead of `any`: `(err as Error).name`. Both `any` and `unknown` may be freely asserted to any type, so it makes pretty good sense to me to assert the type you actually think it is.

Reply:
After talking with Jeremy, for this PR I'm not going to introduce those changes, so I'm discarding them.

Those changes should be done in a different PR when we move to the a newer version of typescript.